src/app/services/baw-api/account.service.spec.ts

```diff
@@ -8,214 +8,378 @@ import { Subject } from "rxjs";
 import { testAppInitializer } from "src/app/app.helper";
 import { User } from "src/app/models/User";
 import { AccountService } from "./account.service";
-import { ApiCommon } from "./api-common";
 import { ApiErrorDetails } from "./api.interceptor";
 import { BawApiService, Filters } from "./base-api.service";
-import { MockApiCommon } from "./mock/api-commonMock";
 import { MockBawApiService } from "./mock/baseApiMockService";
-import { UserService } from "./user.service";
 
 describe("AccountService", () => {
   let httpMock: HttpTestingController;
   let service: AccountService;
 
+  const errorResponse = {
+    status: 401,
+    message: "Unauthorized"
+  } as ApiErrorDetails;
+
+  const errorInfoResponse = {
+    status: 422,
+    message: "Record could not be saved",
+    info: {
+      name: ["has already been taken"],
+      image: [],
+      image_file_name: [],
+      image_file_size: [],
+      image_content_type: [],
+      image_updated_at: []
+    }
+  } as ApiErrorDetails;
+
+  function createError(
+    func:
+      | "apiList"
+      | "apiFilter"
+      | "apiShow"
+      | "apiCreate"
+      | "apiUpdate"
+      | "apiDestroy",
+    url: string,
+    error: ApiErrorDetails
+  ) {
+    spyOn<any>(service as any, func).and.callFake((path: string) => {
+      expect(path).toBe(url);
+      const subject = new Subject();
+
+      setTimeout(() => {
+        subject.error(error);
+      }, 50);
+
+      return subject;
+    });
+  }
+
+  const shouldNotSucceed = () => {
+    fail("Service should not produce a data output");
+  };
+
+  const shouldNotFail = () => {
+    fail("Service should not produce an error");
+  };
+
```

Comment:
these seem sufficiently general enough to be moved to a common test helpers class